### DISCUSSION TITLE
Change the ON during Sequelize LEFT JOIN

### DISCUSSION THREAD

**Discussion by elazsyahranie** [sequelize/sequelize#15452] [Help & Questions] [answered]
Can anybody know how to change the ON during Sequelize LEFT JOIN? 

For example, I'd like to LEFT JOIN Table1 and Table2, but the LEFT JOIN will be ON "Table1.column_1_id" and "Table2.column_2".  For the record, I applied 'hasMany' on 'Table1' and 'belongsTo' on 'Table2'. The code goes something like this: 

// Table1 
Table1.hasMany(models.Table2, { foreignKey: "column_1_id" });

// Table2
Table2.hasMany(models.Table1, { foreignKey: "column_2" }); 

But the problem is, Sequelize always executed `LEFT JOIN on Table1.column_1_id = Table2.column_1_id`. I mean, Sequelize always JOIN on the ID column, which are primary keys in both tables. I wanted to change the way the JOIN is executed on, but no matter what I do, Sequelize always executed `LEFT JOIN on Table1.column_1_id = Table2.column_1_id`. 

Anyone can help?

**Comment by ephys** [comment, marked as answer]
You can configure the other attribute using the `targetKey` option, which defaults to the primary key
